test/unit/gateways/blue_snap_test.rb

```diff
@@ -376,13 +376,58 @@ def test_failed_capture
   end
 
   def test_successful_refund
-    @gateway.expects(:raw_ssl_request).returns(successful_refund_response)
+    options = {
+      reason: 'Refund for order #1992',
+      cancel_subscription: 'false',
+      tax_amount: 0.05,
+      transaction_meta_data: [
+        {
+          key: 'refundedItems',
+          value: '1552,8832',
+          description: 'Refunded Items'
+        },
+        {
+          key: 'Number2',
+          value: 'KTD',
+          description: 'Metadata 2'
+        }
+      ]
+    }
+    transaction_id = '1286'
+    response = stub_comms(@gateway, :ssl_request) do
+      @gateway.refund(@amount, transaction_id, options)
+    end.check_request do |_action, endpoint, data, _headers|
+      doc = REXML::Document.new(data)
 
-    response = @gateway.refund(@amount, 'Authorization')
+      assert_includes endpoint, "/refund/#{transaction_id}"
+      assert_match(/<amount>1.00<\/amount>/, data)
+      assert_match(/<tax-amount>0.05<\/tax-amount>/, data)
+      assert_match(/<cancel-subscription>false<\/cancel-subscription>/, data)
+      assert_match(/<reason>Refund for order #1992<\/reason>/, data)
+      assert_match(/<meta-key>refundedItems<\/meta-key>/, data)
+      assert_match(/<meta-value>KTD<\/meta-value>/, data)
+      assert_match(/<meta-description>Metadata 2<\/meta-description>/, data)
+      transaction_meta_data = doc.root.elements['transaction-meta-data'].elements.to_a
+      transaction_meta_data.each_with_index do |item, index|
+        assert_match item.elements['meta-key'].text, options[:transaction_meta_data][index][:key]
+        assert_match item.elements['meta-value'].text, options[:transaction_meta_data][index][:value]
+        assert_match item.elements['meta-description'].text, options[:transaction_meta_data][index][:description]
+      end
+    end.respond_with(successful_refund_response)
```

Comment:
Thank you for taking to the time to demonstrate that all of these fields end up in the request!